src/java.base/share/classes/java/util/ImmutableCollections.java

```diff
@@ -1582,32 +1636,49 @@ public V getOrDefault(Object key, V defaultValue) {
         }
 
         @jdk.internal.ValueBased
-        final class StableMapEntrySet extends AbstractImmutableSet<Map.Entry<K, V>> {
+        static final class StableMapEntrySet<K, V> extends AbstractImmutableSet<Map.Entry<K, V>> {
+
+            // Use a separate field for the outer class in order to facilitate
+            // a @Stable annotation.
+            @Stable
+            private final StableMap<K, V> outer;
 
             @Stable
             private final Set<Map.Entry<K, StableValueImpl<V>>> delegateEntrySet;
 
-            StableMapEntrySet() {
-                this.delegateEntrySet = delegate.entrySet();
+            private StableMapEntrySet(StableMap<K, V> outer) {
+                this.outer = outer;
+                this.delegateEntrySet = outer.delegate.entrySet();
             }
 
-            @Override public Iterator<Map.Entry<K, V>> iterator() { return new LazyMapIterator(); }
+            @Override public Iterator<Map.Entry<K, V>> iterator() { return LazyMapIterator.of(this); }
             @Override public int                       size() { return delegateEntrySet.size(); }
-            @Override public int                       hashCode() { return StableMap.this.hashCode(); }
+            @Override public int                       hashCode() { return outer.hashCode(); }
 
             @Override
             public String toString() {
                 return StableUtil.renderMappings(this, "StableSet", delegateEntrySet, false);
```

Comment:
```suggestion
                return StableUtil.renderMappings(this, "StableCollection", delegateEntrySet, false);
```
I think `StableList`, `StableSubList`, `StableReverseOrderListView`, `StableMapEntrySet` and `StableMapValues` should all use `"StableCollection"` to be consistent with `AbstractCollection`'s use of `"Collection"`. `StableMap` should use `"StableMap"` to be consistent with `AbstractMap`'s use of `"Map"`.